homeassistant/components/balboa/balboa_entity.py

```diff
@@ -0,0 +1,62 @@
+"""Base class for Balboa Spa Client integration."""
+import time
+
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import _LOGGER, DOMAIN, SPA
+
+
+class BalboaEntity(Entity):
+    """Abstract class for all Balboa platforms.
+
+    Once you connect to the spa's port, it continuously sends data (at a rate
+    of about 5 per second!).  The API updates the internal states of things
+    from this stream, and all we have to do is read the values out of the
+    accessors.
+    """
+
+    _attr_should_poll = False
+
+    def __init__(self, hass, entry, devtype, num=None):
+        """Initialize the spa entity."""
+        self.hass = hass
```

Comment:
Don't set `self.hass`, it will be set when the entity is added via `async_added_to_hass`